include/RAJA/pattern/params/forall.hpp

```diff
@@ -13,6 +18,100 @@ namespace RAJA
 namespace expt
 {
 
+template<typename... Params>
+RAJA_HOST_DEVICE constexpr auto filter_reducers(camp::tuple<Params...>& params)
+{
+  return camp::get_refs_to_elements_by_type_trait<is_instance_of_reducer>(
+      params);
+}
+
+template<typename ExecPol,
+         typename ParamTuple,
+         camp::idx_t... Seq,
+         typename... Args>
+void resolve_params_helper(ParamTuple& params_tuple,
```

Comment:
Should these be in a detail namespace? Once expt is removed they would be directly in RAJA.